### PR TITLE
Map component

### PR DESCRIPTION
**Pull request by sudo-joseph**
This PR implements the following FE functionality. 
1. Ask user for browser location with html 5 geolocation api
2. Map Component displaying mapbox map, centered on user provided lat lon with pan, scroll, & zoom functionality. (Note: Map state stored locally in Map component. Need to evaluate if this needs to be moved to global state, which may be somewhat challenging). 
3. Modal open on geolocation API rejection or failure. 
4. Basic Card Component.
5. App level state.

### CODE REVIEW

**Comment by sudo-joseph** [comment]
Current Recycle Page: 
image

Current Modal View: 
image

**Comment by sudo-joseph** [comment]
Fixed merge conflicts.

File: client/src/App.js [hunk 1]
```diff
@@ -10,27 +10,87 @@ import NavBar from './components/NavBar/NavBar.js';
...
+              <Link to="/recycle/">Recycle</Link>
+            </nav>
+
+            <div className="App-mainContent">
+              <Switch>
+                <Route exact="exact" path='/' component={Welcome}/>
+                <Route exact="exact" path='/browse/' component={Browse}/>
+                <Route exact="exact"
+                       path='/recycle/'
+                       render={(routeProps) => (<RecyclePage {...routeProps}
```
Comment (ishinan) [inline client/src/App.js:82]: routeProps was new to me. I learned a coding :)

File: client/src/components/Card/Card.js [hunk 2]
```diff
@@ -1,13 +1,22 @@
-import React, { Component } from 'react';
+import React, {Component} from 'react';
```
Comment (ishinan) [inline client/src/components/Card/Card.js:1]: F.Y.I. `{ Component }`, spaces between functions/classes in import statement is more common.

File: client/src/components/Map/Map.js [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
+  }
+
   render() {
 
-    return (
-      <div className="Map-Title">
-      </div>
-    );
+    return (<div className="Map-Title">
+      <div ref={el => this.mapContainer = el} className="mapContainer"/>
```
Comment (ishinan) [inline client/src/components/Map/Map.js:29]: Wow, this is how to set a ref. Confusing but interesting.

**Review by ishinan — APPROVED** [review]
Great start. This PR already achieved creating Map component. I love it.

**Review by VeniVidiCodi — APPROVED** [review]
Looks great. Really great work.

File: client/src/components/Card/Card.js (hunk 2 above)
Comment (sudo-joseph) [reply client/src/components/Card/Card.js:1]: oh thanks.

File: client/src/components/Map/Map.js (hunk 3 above)
Comment (sudo-joseph) [reply client/src/components/Map/Map.js:29]: yeah. thankfully mapbox had a really good tutorial

File: client/src/App.js (hunk 1 above)
Comment (sudo-joseph) [reply client/src/App.js:82]: oh yeah this was confusing. we should have gone over this in class imo

**Pull request merged**